pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -144,6 +145,92 @@ func (r *ReconcileComplianceScan) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileComplianceScan) handleRootCASecret() error {
+	secret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: "compliance-operator-root-ca", Namespace: "openshift-compliance"}, secret)
+	if err == nil {
+		// Already exists, so return.
+		return nil
+	}
+	if !errors.IsNotFound(err) {
+		return err
+	}
+
+	// Create the CA.
+	cert, key, err := utils.ComplianceOperatorRootCA("compliance-operator-root-ca", 100)
+	if err != nil {
+		return err
+	}
+
+	secret = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "compliance-operator-root-ca",
+			Namespace: "openshift-compliance",
+		},
+		Data: map[string][]byte{
+			"ca.crt": cert,
+			"ca.key": key,
+		},
+	}
+
+	// Create the CA secret.
+	err = r.client.Create(context.TODO(), secret)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
+	}
+	// Do not create controlle references because for now, we want the CA to be created first scan, and then
+	// remain for subsequent scans. Might help with race conditions for server/client certs.
+	// Force rotate with a secret deletion.
```

Comment:
So the rotation would be up to the admin?

Reply:
This is a good point to bring up. Rotation should be fairly simple (as Matt put out) and should be a matter of deleting the Secret. Should we have a process that does this daily or weekly as long as there isn't a scan running?

The certs that the CA gives out are only relevant in the context of scans (to upload results to the resultserver) which is deleted when the scan is in phase DONE.

Reply:
We can have the controller check the expiration and regenerate the CA, right now I just have a check for if it exists.  We could then use something short-lived, like around what we think might be the maximum for a single scan (1hr?). Then the admin doesn't need to know anything.

Reply:
That sounds good to me. Regenerating the CA past some percentage would still probably be a good idea, because otherwise you might have a second scans scheduled 59mins after the first one and run out of the lifetime..

Reply:
I'm not sure if the new CA per scan or a limited use one would be better. For the moment, I updated to make it so there is now one CA per scan (cleaned up by CR reference) and they are valid for a day.  (The library-go helpers only give me a day as minimum) .